crates/conjure_core/src/rule_engine/rewrite.rs

```diff
@@ -347,6 +348,18 @@ fn apply_all_rules<'a>(
 /// }
 ///
 fn choose_rewrite(results: &[RuleResult]) -> Option<Reduction> {
+    //Uncomment the following if want to check for mutual exclusivity of rules
+    // if results.len() > 1 {
+    //     let expr = results[0].reduction.new_expression.clone();
```

Comment:
I think that we should report the expression pre-rewrite

Reply:
let's do this change and reenable the check @Soph1514?

Reply:
> 

Should I uncomment it? The tests are still failing